contracts/RelayHub.sol

```diff
@@ -135,19 +142,24 @@ contract RelayHub is IRelayHub {
             IPaymaster(relayRequest.relayData.paymaster).getGasAndDataLimits{gas:50000}();
         require(msg.data.length <= gasAndDataLimits.calldataSizeLimit, "msg.data exceeded limit" );
         uint256 dataGasCost = calldataGasCost(msg.data.length);
+        uint256 txDataCost = externalGasLimit - initialGasLeft - relayCallDataOverhead;
+        uint256 txDataCostPerByte = txDataCost/msg.data.length;
+        require(txDataCostPerByte >= G_ZERO && txDataCostPerByte <= G_NONZERO, "invalid externalGasLimit");
+
         require(maxRelayExposure >= gasAndDataLimits.acceptanceBudget.add(dataGasCost), "pm budget + dataGasCost too high");
 
         maxPossibleGas =
             gasOverhead.add(
             gasAndDataLimits.preRelayedCallGasLimit).add(
             gasAndDataLimits.postRelayedCallGasLimit).add(
             relayRequest.request.gas).add(
-            dataGasCost);
+            dataGasCost).add(
+            txDataCost);
```

Comment:
these two names are confusing, let's look for a better couple, maybe?

Reply:
onChainDataCost and externalDataCost ?

Reply:
good enough